dm/syncer/checkpoint.go

```diff
@@ -154,8 +154,8 @@ func (b *binlogPoint) outOfDate() bool {
 func (b *binlogPoint) outOfDateBy(pos binlog.Location) bool {
 	b.RLock()
 	defer b.RUnlock()
-
-	return binlog.CompareLocation(pos, b.flushedPoint.location, b.enableGTID) > 0
+	// locations in savedPoint have no suffix and may have same giid like handle-error, we can use position to check them
+	return binlog.CompareLocation(pos, b.flushedPoint.location, false) > 0
```

Comment:
seems change the logic in `binlog.CompareLocation` is more reasonable ?, hardcode to enable gtid to false is too tricky

Reply:
`binlog.CompareLocation` is called in many places and i think we may fix it when we refine suffix. this is `outOfDateBy` is used to check between table checkpoint and global checkpoint for flush, this kind of point should have right position, so  we can use position to check.

Reply:
>  add more tests about inject

will you doing this in thie pr or, is this pr ready for review?

Reply:
not yet. after some test, it will be more complex when use sharding ddl.